packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_amp.yml

```
@@ -9,7 +9,7 @@ processors:
         - "^File reputation query initiating. %{GREEDYDATA:_tmp.new_message}$"
         - "^Response received for file reputation query from Cloud. %{GREEDYDATA:_tmp.new_message}$"
         - "^File Analysis complete. SHA256: %{GREEDYDATA:email.attachments.file.hash.sha256}, Submit Timestamp: %{GREEDYDATA:_tmp.submit.timestamp}, Update Timestamp: %{GREEDYDATA:_tmp.update.timestamp}, Disposition: %{DATA:cisco_secure_email_gateway.log.disposition} Score: %{NUMBER:cisco_secure_email_gateway.log.score:long}, run_id: %{NUMBER:cisco_secure_email_gateway.log.run_id} Details: %{DATA:cisco_secure_email_gateway.log.details} Spyname:\\[%{GREEDYDATA:cisco_secure_email_gateway.log.spy_name}\\]$"
-        - "^File not uploaded for analysis. MID = %{NUMBER:email.message_id} File SHA256\\[%{GREEDYDATA:email.attachments.file.hash.sha256}\\] file mime\\[%{GREEDYDATA:email.attachments.file.mime_type}\\] Reason: %{GREEDYDATA:event.reason}$"
+        - "^File not uploaded for analysis.\\s+MID = %{NUMBER:email.message_id}, File SHA256\\[%{GREEDYDATA:email.attachments.file.hash.sha256}\\], File mime\\[%{GREEDYDATA:email.attachments.file.mime_type}\\], Reason: %{GREEDYDATA:event.reason}$"
```

```suggestion
        - "^(?i)File not uploaded for analysis.\\s+MID = %{NUMBER:email.message_id},? File SHA256\\[%{GREEDYDATA:email.attachments.file.hash.sha256}\\],? File mime\\[%{GREEDYDATA:email.attachments.file.mime_type}\\],? Reason: %{GREEDYDATA:event.reason}$"
```
(I agree that the post-MID comma is probably in all messages, but since we have an example without, unless we can definitively show that it never exists, we should make it optional, same for the post-MIME type comma - see https://quickview.cloudapps.cisco.com/quickview/bug/CSCva85147 for an example that matches the current pattern)

---

Another issue is that the _File_  word before _mime_ will be capitalized if there are commas and not capitalized if there are no commas, I think that adding `(F|f)` will catch both, will fix the test and grok pattern to catch both types of messages.

---

Oh, nevermind, just saw the `(?i)` to deal with the cases.

---

Yeah, also, generally prefer `[Ff]` over `(F|f)` if we weren't using` `(?i)`.